src/Microsoft.Identity.Core/CoreErrorMessage.cs

```diff
@@ -0,0 +1,34 @@
+﻿//----------------------------------------------------------------------
+//
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+//
+// This code is licensed under the MIT License.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+//------------------------------------------------------------------------------
+
+namespace Microsoft.Identity.Core
+{
+    internal static class CoreErrorMessage
+    {
+        public const string Unknown = "Unknown error";
```

Comment:
This seems like a duplicate (and not used) message which is defined and used in CoreException?

Reply:
in Core Exception error code is defined, not error mesage

Reply:
what is really the difference? Why is this needed?

---
In reply to: [160310180](https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/pull/925#discussion_r160310180) [](ancestors = 160310180)

Reply:
error codes represents general errors where message is details about it (it can be string const, or template filled with values in real time or in can be response from sts ...)  Also all Exceptions are part of public api, so  we might have customers who have dependencies on  code and message properties changing structure of exception will be breaking changes.

Reply:
I don't get it... sorry. to me these two things belongs together.